src/main/java/org/folio/service/file/upload/FileUploadServiceImpl.java

```diff
@@ -151,16 +147,18 @@
 
   private Future<FileDefinition> uploadFileWithCQLQueryQuickExport(QuickExportRequest request, FileDefinition fileDefinition, JobExecution jobExecution, OkapiConnectionParams params) {
     Promise<FileDefinition> promise = Promise.promise();
-    Optional<JsonObject> instancesUUIDs = inventoryClient.getInstancesBulkUUIDs(request.getCriteria(), params);
-    if (instancesUUIDs.isPresent()) {
-      saveUUIDsFromJson(instancesUUIDs.get(), fileDefinition, jobExecution, request, params)
-        .onSuccess(promise::complete)
-        .onFailure(ar -> failFileDefinitionAndJobExecution(promise, fileDefinition, jobExecution, request, ar.getCause(), params));
-    } else {
-      saveUUIDsFromJson(new JsonObject(), fileDefinition, jobExecution, request, params)
-        .onSuccess(promise::complete)
-        .onFailure(ar -> failFileDefinitionAndJobExecution(promise, fileDefinition, jobExecution, request, ar.getCause(), params));
-    }
+    inventoryClient.getInstancesBulkUUIDsAsync(request.getCriteria(), params).onComplete(instancesUUIDsResult -> {
+      Optional<JsonObject> instancesUUIDs = instancesUUIDsResult.result();
```

Comment:
Here the Optional object always will be returned, no need into onSuccess/onFailure methods